flight/src/tests.rs

```diff
@@ -0,0 +1,107 @@
+                )
+                .unwrap(),
+                stream_type: StreamIdType::Model as u8,
+                controller: "did:key:test3".to_string(),
+                dimensions: vec![],
+            },
+        }),
+        ConclusionEvent::Data(ConclusionData {
+            event_cid: Cid::from_str("baeabeials2i6o2ppkj55kfbh7r2fzc73r2esohqfivekpag553lyc7f6bi")
+                .unwrap(),
+            init: ConclusionInit {
+                stream_cid: Cid::from_str(
+                    "baeabeif2fdfqe2hu6ugmvgozkk3bbp5cqi4udp5rerjmz4pdgbzf3fvobu",
+                )
+                .unwrap(),
+                stream_type: StreamIdType::Model as u8,
+                controller: "did:key:test4".to_string(),
+                dimensions: vec![],
+            },
+            previous: vec![],
+            data: vec![7, 8, 9],
+        }),
+    ];
+    // Convert events to RecordBatch
+    let record_batch = conclusion_events_to_record_batch(&events).unwrap();
+
+    // Convert RecordBatch to string
+    let formatted = pretty_format_batches(&[record_batch.clone()])
+        .unwrap()
+        .to_string();
+
+    // Use expect_test to validate the output
+    expect![[r#"
+        +------------+--------------------------------------------------------------------------+-------------+---------------+--------------------------------------------------------------------------+--------+----------------------------------------------------------------------------+
+        | event_type | stream_cid                                                               | stream_type | controller    | event_cid                                                                | data   | previous                                                                   |
+        +------------+--------------------------------------------------------------------------+-------------+---------------+--------------------------------------------------------------------------+--------+----------------------------------------------------------------------------+
+        | 0          | 01001220ba28cb0268f4f50cca99d952b610bfa2823941bfb12452ccf1e330725d96ae0d | 2           | did:key:test1 | 010012200b9691e769ef527bd51427fc745c8bfb8e89271e054548a780ddeed7817cbe0a | 010203 | []                                                                         |
+        | 0          | 01001220ba28cb0268f4f50cca99d952b610bfa2823941bfb12452ccf1e330725d96ae0d | 2           | did:key:test2 | 010012200b9691e769ef527bd51427fc745c8bfb8e89271e054548a780ddeed7817cbe0a | 040506 | [01001220ba28cb0268f4f50cca99d952b610bfa2823941bfb12452ccf1e330725d96ae0d] |
+        | 1          | 01001220ba28cb0268f4f50cca99d952b610bfa2823941bfb12452ccf1e330725d96ae0d | 2           | did:key:test3 | 010012200b9691e769ef527bd51427fc745c8bfb8e89271e054548a780ddeed7817cbe0a |        |                                                                            |
```

Comment:
The previous column is empty here, however the input data does include a previous cid. This is caused by the bug I noted above.